test/VersionPropsFormatter.Tests/VersionPropsFormatterTest.cs

```diff
@@ -0,0 +1,132 @@
+
+    private const string VersionDetails = """
+        <Dependencies>
+          <ProductDependencies>
+            <Dependency Name="Foo" Version="1.0.0">
+              <Uri>https://github.com/dotnet/foo</Uri>
+              <Sha>sha1</Sha>
+            </Dependency>
+            <Dependency Name="Foo2" Version="2.0.0">
+              <Uri>https://github.com/dotnet/foo</Uri>
+              <Sha>sha1</Sha>
+            </Dependency>
+            <Dependency Name="Bar" Version="1.0.0">
+              <Uri>https://github.com/dotnet/bar</Uri>
+              <Sha>sha1</Sha>
+            </Dependency>
+          </ProductDependencies>
+          <ToolsetDependencies>
+          </ToolsetDependencies>
+        </Dependencies>
+        """;
+
+    private const string VersionProps = """
+        <?xml version="1.0" encoding="utf-8"?>
+        <Project>
+          <PropertyGroup>
+          </PropertyGroup>
+          <!--Package versions-->
+          <PropertyGroup>
+            <FooPackageVersion>1.0.0</FooPackageVersion>
+            <BarPackageVersion>1.0.0</BarPackageVersion>
+          </PropertyGroup>
+        </Project>
+        """;
+    private const string ExpectedWarning = """
+    The following dependencies were found in Version.Details.xml, but not in Version.props. Consider removing them
+      Foo2
+    """;
+    private const string ExpectedOutput = """
+        <PropertyGroup>
+          <!-- Foo dependencies -->
+          <FooPackageVersion>1.0.0</FooPackageVersion>
+          <!-- Bar dependencies -->
+          <BarPackageVersion>1.0.0</BarPackageVersion>
+        </PropertyGroup>
+        """;
+
+    [SetUp]
+    public void Setup()
+    {
+        IServiceCollection serviceCollection = new ServiceCollection();
+        VersionPropsFormatter.RegisterServices(serviceCollection);
+        _loggerMock = new Mock<ILogger<VersionPropsFormatter>>();
+        serviceCollection.AddSingleton(_loggerMock.Object);
+
+        _serviceProvider = serviceCollection.BuildServiceProvider();
+    }
+
+    [Test]
+    public async Task TestVersionPropsFormatter()
+    {
+        NativePath tmpFolder = new(Path.Combine(Path.GetTempPath(), Path.GetTempFileName()));
+        IProcessManager processManager = _serviceProvider.GetRequiredService<IProcessManager>();
+
+        try
+        {
+            if (Directory.Exists(tmpFolder))
+            {
+                Directory.Delete(tmpFolder, true);
+            }
+            if (File.Exists(tmpFolder))
+            {
+                File.Delete(tmpFolder);
+            }
+            Directory.CreateDirectory(tmpFolder);
+            await processManager.ExecuteGit(tmpFolder, "init");
+            Directory.CreateDirectory(tmpFolder / Constants.EngFolderName);
+
+            File.WriteAllText(tmpFolder / VersionFiles.VersionDetailsXml, VersionDetails);
+            File.WriteAllText(tmpFolder / VersionFiles.VersionProps, VersionProps);
```

Comment:
the test is failing with this error `Microsoft.DotNet.DarcLib.DependencyFileNotFoundException : Could not find eng/Versions.props in C:\Users\cloudtest\AppData\Local\Temp\tmpz3h2m5.tmp
` but it's not happening locally, need to investigate this further.. perhaps something is wrong with the repo initialization below